osc-ui/src/main/java/org/osc/core/broker/view/maintenance/InternalCertReplacementUploader.java

```diff
@@ -52,11 +59,35 @@ protected void createUpload() {
         this.upload.setDescription(getString(KEYPAIR_UPLOAD_DESCR));
     }
 
+    @SuppressWarnings("serial")
     @Override
     protected void processCertificateFile(File file) throws Exception {
         log.info("================ SSL certificate upload completed");
         log.info("================ Replacing internal certificate in truststore...");
-        this.x509TrustManager.replaceInternalCertificate(file, true);
-        removeUploadedFile();
+
+        final VmidcWindow<OkCancelButtonModel> alertWindow =
+                WindowUtil.createAlertWindow("Warning", getString(KEYPAIR_UPLOAD_WARN_CONFIRM));
+
+        alertWindow.getComponentModel().setOkClickedListener(new Button.ClickListener(){
```

Comment:
create two private methods setOkClickedListener and setCancelClickedListener. Both receiving the alertWindow. This will make the processCertificateFile method smaller and easier, easier to read.